Lib/test/test_capi/test_misc.py

```diff
@@ -1098,6 +1098,55 @@ class Data(_testcapi.ObjExtraData):
         del d.extra
         self.assertIsNone(d.extra)
 
+    def test_sys_getattr(self):
+        sys_getattr = _testcapi.sys_getattr
+
+        self.assertIs(sys_getattr('stdout'), sys.stdout)
+        with support.swap_attr(sys, '\U0001f40d', 42):
+            self.assertEqual(sys_getattr('\U0001f40d'), 42)
+
+        with self.assertRaisesRegex(RuntimeError, r'lost sys\.nonexisting'):
+            sys_getattr('nonexisting')
+        with self.assertRaisesRegex(RuntimeError, r'lost sys\.1'):
+            sys_getattr(1)
+        self.assertRaises(TypeError, sys_getattr, [])
+        # CRASHES sys_getattr(NULL)
+
+    def test_sys_getattrstring(self):
+        sys_getattr = _testcapi.sys_getattrstring
+
+        self.assertIs(sys_getattr(b'stdout'), sys.stdout)
+        with support.swap_attr(sys, '\U0001f40d', 42):
+            self.assertEqual(sys_getattr('\U0001f40d'.encode()), 42)
+
+        with self.assertRaisesRegex(RuntimeError, r'lost sys\.nonexisting'):
+            sys_getattr(b'nonexisting')
+        self.assertRaises(UnicodeDecodeError, sys_getattr, b'\xff')
+        # CRASHES sys_getattr(NULL)
+
+    def test_sys_getoptionalattr(self):
+        sys_getattr = _testcapi.sys_getoptionalattr
```

Comment:
It's surprising that in all tests, the function is called "sys_getattr", whereas here you test PySys_GetOptionalAttr(), not PySys_GetAttr().

I suggest to rename the variable t o"sys_getoptionalattr", or even PySys_GetOptionalAttr() since this is a C API test. It would be more explicit to use the name of the C API.